### ISSUE TITLE
VST3 Trigger MIDI not sending events.

### ISSUE THREAD

**Issue by Stazed** [lsp-plugins/lsp-plugins#410]
A little debugging, I checked the MIDI out processing for VST3 using the Sampler plugin MIDI pass-through.
The Sampler MIDI out and host seem to correctly process the MIDI out pass-through.

In file:
```
lsp-plugins/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/wrap/vst3/impl/wrapper.h
```
In function: Line 2458:
```
void Wrapper::process_output_events(Steinberg::Vst::IEventList *events)
```

Line 2467: It looks like pEventsIn should be pEventsOut.
```
-- vst3::MidiPort *p = pEventsIn->vPorts[i];
```
```
++ vst3::MidiPort *p = pEventsOut->vPorts[i];
```
Doing the above change fixes the Trigger MIDI, but breaks the Sampler pass-through.
The above change caused the pass-through to constantly repeat Note on/ Note off messages.

So to fix the pass-through after the above change I added a queue->clear() after the
Line 2238: plug::midi_t *queue = p->queue();

...in the void Wrapper::process_input_events(){} function.

Both Sampler pass-through and Trigger MIDI plugins MIDI out seem to work after the above.
Also, the queue->clear() did NOT seem to break the Sampler Note On to trigger sample playing.
I'm not sure it is the correct way to fix it, but hopefully it will help narrow things down.

**Comment by Stazed** [comment]
Changing the typo breaks MIDI pass-through of the Sampler plugins. Kind of strange, but this is because pass-through worked accidentally because of the typo by calling queue->clear() on pEventsIn port. By fixing the typo, pass-through now never clears the queue for pEventsIn and any Notes will get repeatedly sent to the output MIDI port for pass-through of the Sampler plugins.

>So to fix the pass-through after the above change I added a queue->clear() after the
Line 2238: plug::midi_t *queue = p->queue();

>...in the void Wrapper::process_input_events(){} function.

Weird, but check the Sampler MIDI out pass-through events after the typo fix and you will see what I mean...

**Comment by sadko4u** [comment]
I think we just need to clear the queue at the start of processing. I'll add some code now..

**Comment by sadko4u** [comment]
This should fix the problem: https://github.com/lsp-plugins/lsp-plugin-fw/commit/f2634cec3c1b5af511b672b5054060abf90b3a8d

**Comment by Stazed** [comment]
Yes that fixes it. Thanks.

**Comment by sadko4u** [comment]
Fixed in 1.2.17
